src/components/ToolBar.js

```diff
@@ -0,0 +1,46 @@
+var React = require('react');
+var Events = require('../lib/Events.js');
+
+var TransformButtons = [
+  {value: 'translate', icon: 'fa-arrows'},
+  {value: 'rotate', icon: 'fa-repeat'},
+  {value: 'scale', icon: 'fa-expand'}
+];
+
+export default class ToolBar extends React.Component {
+  constructor (props) {
+    super(props);
+    this.state = {selectedTransform: 'translate', localSpace: false};
+  }
+
+  changeTransformMode = mode => {
+    this.setState({selectedTransform: mode});
+    Events.emit('transformModeChanged', mode);
+  }
+
+  onLocalChange = e => {
+    var local = e.target.checked;
+    this.setState({localSpace: local});
+    Events.emit('spaceChanged', local ? 'local' : 'world');
+  }
+
+  render () {
+    return <div className='toolbar'>
+      {
```

Comment:
I usually try to keep code blocks out of the render to be more readable.

```js
renderTransformButtons = () => {
  return TransformButtons.map  // ...
}
```

The fat arrow will also do the `this` binding for you.